From 5c9f464ab89a634dbc74369ac29c7c6e456fa5c8 Mon Sep 17 00:00:00 2001
From: Niko Tyni <ntyni@debian.org>
Date: Tue, 29 Jul 2014 15:21:29 +0300
Subject: [PATCH 2/2] Add compatibility with Perl 5.20

Two error messages have changed slightly, dropping the 'Might be a
multi-line string' notes.

According to https://rt.cpan.org/Public/Bug/Display.html?id=95335
this is due to upstream commit v5.19.3-296-gffdb8b1 so the conditionals
now look for 5.19.4 although this has only been tested with 5.20.0.

Bug: https://rt.cpan.org/Public/Bug/Display.html?id=95335
Bug-Debian: https://bugs.debian.org/754997
---
 test.pl                     |  9 +++++-
 test/cmp/includeerr2.htm520 | 69 +++++++++++++++++++++++++++++++++++++++++++++
 2 files changed, 77 insertions(+), 1 deletion(-)
 create mode 100644 test/cmp/includeerr2.htm520

diff --git a/test.pl b/test.pl
index 9714453..f19445e 100644
--- a/test.pl
+++ b/test.pl
@@ -435,9 +435,16 @@
         'errors'     => 9,
         'version'    => 2,
         'repeat'     => 2,
-        'condition'  => '$] >= 5.018000', 
+        'condition'  => '$] >= 5.018000 && $] < 5.019400', 
         'cmpext'     => '518',
         },
+    'includeerr2.htm' => { 
+        'errors'     => 7,
+        'version'    => 2,
+        'repeat'     => 2,
+        'condition'  => '$] >= 5.019400', 
+        'cmpext'     => '520',
+        },
     'includeerr3.htm' => { 
         'errors'     => 2,
         'condition'  => '$] < 5.014000', 
diff --git a/test/cmp/includeerr2.htm520 b/test/cmp/includeerr2.htm520
new file mode 100644
index 0000000..a755c10
--- /dev/null
+++ b/test/cmp/includeerr2.htm520
@@ -0,0 +1,69 @@
+<HTML><HEAD><TITLE>Embperl Error</TITLE></HEAD><BODY bgcolor="#FFFFFF">
+<H1>Internal Server Error</H1>
+The server encountered an internal error or misconfiguration and was unable to complete your request.<P>
+^Please contact the server administrator,.*?and inform them of the time the error occurred, and anything you might have done that may have caused the error.<P><P>
+<table cellspacing='2' cellpadding='5'>
+<tr bgcolor='#eeeeee'><td>
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+^\[\d+\]ERR:  32:  Warning in Perl code: Use of uninitialized value \$Embperl::__\d+::param\[0\] in concatenation \(\.\) or string at .+incsub.htm line 6.
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+</td></tr>
+<tr bgcolor='#eeeeee'><td>
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+^\[\d+\]ERR:  32:  Warning in Perl code: Use of uninitialized value \$Embperl::__\d+::param\[0\] in concatenation \(\.\) or string at .+incsub.htm line 6.
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+</td></tr>
+<tr bgcolor='#eeeeee'><td>
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+^\[\d+\]ERR:  32:  Warning in Perl code: Use of uninitialized value \$Embperl::__\d+::param\[0\] in concatenation \(\.\) or string at .+incsub.htm line 6.
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+</td></tr>
+<tr bgcolor='#eeeeee'><td>
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+^\[\d+\]ERR:  32:  Warning in Perl code: Use of uninitialized value \$Embperl::__\d+::param\[0\] in concatenation \(\.\) or string at .+incsub.htm line 6.
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+</td></tr>
+<tr bgcolor='#eeeeee'><td>
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+^\[\d+\]ERR:  24:  Error in Perl code: Can't locate object method &quot;is&quot; via package &quot;here&quot; \(perhaps you forgot to load &quot;here&quot;\?\) at .+incerr.htm line 6.
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+</td></tr>
+<tr bgcolor='#eeeeee'><td>
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+^\[\d+\]ERR:  32:  Warning in Perl code: Unquoted string &quot;table&quot; may clash with future reserved word at .+incerr.htm line 6.
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+</td></tr>
+<tr bgcolor='#eeeeee'><td>
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+^\[\d+\]ERR:  32:  Warning in Perl code: Unquoted string &quot;td&quot; may clash with future reserved word at .+incerr.htm line 8.
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+</td></tr>
+<tr bgcolor='#eeeeee'><td>
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+^\[\d+\]ERR:  24:  Error in Perl code: syntax error at .+line 4, near &quot;\+\]&quot;
+^<br>&nbsp;&nbsp;&nbsp;&nbsp;Unknown regexp modifier &quot;\/t&quot; at 
+^<br>&nbsp;&nbsp;&nbsp;&nbsp;Unknown regexp modifier &quot;\/r&quot; at 
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+</td></tr>
+<tr bgcolor='#eeeeee'><td>
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+^\[\d+\]ERR:  32:  Warning in Perl code: Unquoted string &quot;table&quot; may clash with future reserved word at .+incerr.htm line 6.
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+</td></tr>
+<tr bgcolor='#eeeeee'><td>
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+^\[\d+\]ERR:  32:  Warning in Perl code: Unquoted string &quot;td&quot; may clash with future reserved word at .+incerr.htm line 8.
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+</td></tr>
+<tr bgcolor='#eeeeee'><td>
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+^\[\d+\]ERR:  24:  Error in Perl code: syntax error at .+line 4, near &quot;\+\]&quot;
+^<br>&nbsp;&nbsp;&nbsp;&nbsp;Unknown regexp modifier &quot;\/t&quot; at 
+^<br>&nbsp;&nbsp;&nbsp;&nbsp;Unknown regexp modifier &quot;\/r&quot; at 
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+</td></tr>
+</table>
+<br>
+^Embperl
+</BODY></HTML>
+
-- 
2.0.1

